web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/ComplementCertificateFromCertificateService.java

```diff
@@ -0,0 +1,107 @@
+import se.inera.intyg.webcert.web.service.monitoring.MonitoringLogService;
+import se.inera.intyg.webcert.web.service.user.WebCertUserService;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service("complementCertificateFromCertificateService")
+public class ComplementCertificateFromCertificateService implements ComplementCertificateFacadeService {
+
+    private final CSIntegrationService csIntegrationService;
+    private final CSIntegrationRequestFactory csIntegrationRequestFactory;
+    private final WebCertUserService webCertUser;
+    private final MonitoringLogService monitoringLogService;
+    private final IntegratedUnitRegistryHelper integratedUnitRegistryHelper;
+    private final PublishCertificateStatusUpdateService publishCertificateStatusUpdateService;
+    private final PDLLogService pdlLogService;
+
+    @Override
+    public Certificate complement(String certificateId, String message) {
+        log.debug("Attempting to complement certificate '{}' from Certificate Service", certificateId);
+
+        if (Boolean.FALSE.equals(csIntegrationService.certificateExists(certificateId))) {
+            log.debug("Certificate '{}' does not exist in certificate service", certificateId);
+            return null;
+        }
+
+        final var certificateToComplement = csIntegrationService.getCertificate(
+            certificateId,
+            csIntegrationRequestFactory.getCertificateRequest()
+        );
+
+        final var certificate = csIntegrationService.complementCertificate(
+            certificateId,
+            csIntegrationRequestFactory.complementCertificateRequest(
+                certificateToComplement.getMetadata().getPatient(),
+                webCertUser.getUser().getParameters(),
+                message)
+        );
+
+        integratedUnitRegistryHelper.addUnitForCopy(certificateToComplement, certificate);
+
+        log.debug("Complemented certificate '{}' from Certificate Service", certificateId);
+        monitoringLogService.logIntygCopiedCompletion(
+            certificate.getMetadata().getId(),
+            certificateToComplement.getMetadata().getId()
+        );
+        pdlLogService.logCreated(certificate);
+        publishCertificateStatusUpdateService.publish(certificate, HandelsekodEnum.SKAPAT);
+
+        return certificate;
+    }
+
+    @Override
+    public Certificate answerComplement(String certificateId, String message) {
+        log.debug("Attempting to answer complement on certificate '{}' from Certificate Service", certificateId);
+
+        if (Boolean.FALSE.equals(csIntegrationService.certificateExists(certificateId))) {
+            log.debug("Certificate '{}' does not exist in certificate service", certificateId);
+            return null;
+        }
+
+        final var certificate = csIntegrationService.answerComplementOnCertificate(
+            certificateId,
+            csIntegrationRequestFactory.answerComplementOnCertificateRequest(message)
+        );
+
```

Comment:
We should send a status update for the certificate that an answer has been sent. See ArendeServiceImpl.answerKomplettering